public/app.js

```diff
@@ -56,11 +83,17 @@ class Controller {
         Controller.setupEventListeners();
         // Add the dropdown menu to the UI
         DataController.watchActiveProjects(UIController.addDropdownMenu);
-        // Add the coding progress section to the UI
-        UIController.addCodingProgressSection();
-        // Get data for coding progress table
-        let unsubscribeFunc = DataController.watchCodingProgress(UIController.updateProgressUI);
-        DataController.registerSnapshotListener(unsubscribeFunc);
+        // Get hash value
+        let page_route = window.location.hash.substring(1);
+        // Navigate appropriately according to the hash value
+        if (page_route) {
+            if (page_route == "coding_progress") {
+                Controller.displayCodingProgress();
+            }
+            DataController.watchActiveProjects(Controller.displayDeepLinkedTrafficPage);
```

Comment:
This isn't safe, there's no guarantee that this is a valid traffic hash.

Consider:
```javascript
if (page_route == "coding_progress") {
    ...
} else if (page_route.startswith("traffic-") {
   ...
} else {
    ...
}
```
or use a switch.